OrbitCore/LinuxTracingBuffer.cpp

```diff
@@ -138,6 +138,38 @@ bool LinuxTracingBuffer::ReadAllThreadNames(
   return true;
 }
 
+bool LinuxTracingBuffer::IsEmpty() {
+  {
+    absl::MutexLock lock(&context_switch_buffer_mutex_);
+    if (!context_switch_buffer_.empty()) return false;
+  }
+  {
+    absl::MutexLock lock(&timer_buffer_mutex_);
+    if (!timer_buffer_.empty()) return false;
+  }
+  {
+    absl::MutexLock lock(&callstack_buffer_mutex_);
+    if (!callstack_buffer_.empty()) return false;
+  }
+  {
+    absl::MutexLock lock(&hashed_callstack_buffer_mutex_);
+    if (!hashed_callstack_buffer_.empty()) return false;
+  }
+  {
+    absl::MutexLock lock(&address_info_buffer_mutex_);
+    if (!address_info_buffer_.empty()) return false;
+  }
+  {
+    absl::MutexLock lock(&key_and_string_buffer_mutex_);
+    if (!key_and_string_buffer_.empty()) return false;
+  }
+  {
+    absl::MutexLock lock(&thread_name_buffer_mutex_);
+    if (!thread_name_buffer_.empty()) return false;
+  }
+  return true;
```

Comment:
This is not really thread-safe.. Since it seems this is only used for CHECK call maybe just drop it? Add a test instead?

Reply:
I'd be fine with dropping it. Still not sure, how such a test should look like.

Reply:
This seems like a fairly complex piece of code for adding a CHECK. I would also drop it completely. It's complex enough that you could have bugs in the check here (and as @dimitry- noted you do have a bug, the result of returning "true" may be incorrect by the time you get there if the buffers are accessed from multiple threads).

Reply:
I just updated the PR with dropping the check. Still not 100% sure, how to write a reasonable test for this.

Reply:
I do not think we already have a test for that. It tests that after readall the buffer is empty - I do not think there is more to test.https://github.com/google/orbit/blob/master/OrbitCore/LinuxTracingBufferTest.cpp

Reply:
Added the expectations to all the tests.

Reply:
As discussed offline, this is already tested by the contract of read and the actual buffers should remain private to the tests, as they are impl. detailes. So I reverted the change.